src/openforms/js/components/admin/form_design/logic/Trigger.js

```diff
@@ -197,64 +202,62 @@ const Trigger = ({ name, logic, onChange }) => {
         ],
     };
 
-    const previousLogicFromState = usePrevious(jsonLogicFromState);
-
     // whenever we get a change in the jsonLogic definition, relay that back to the
     // parent component
-    useEffect(
-        () => {
-            // if nothing changed, do not fire an update
-            if (previousLogicFromState && isEqual(previousLogicFromState, jsonLogicFromState)) return;
-            onChange({
-                target: {
-                    name: name,
-                    value: jsonLogicFromState,
-                }
-            });
-        },
-        [jsonLogicFromState]
+    useOnChanged(
+        jsonLogicFromState,
+        () => onChange({target: {name, value: jsonLogicFromState}})
     );
 
     return (
-        <div style={{padding: '1em'}}>
-
-            When
-
-            <div>
-                <ComponentSelection
-                    name="component"
-                    value={triggerComponent}
-                    onChange={onTriggerChange}
-                />
-                &nbsp;
-                {
-                    triggerComponent ? (
-                        <OperatorSelection
-                            name="operator"
-                            selectedComponent={triggerComponent}
-                            operator={operator}
+        <div className="logic-trigger">
+            <div className="logic-trigger__editor">
+                <div className="dsl-editor">
+                    <div className="dsl-editor__node">
+                        <FormattedMessage description="Logic trigger prefix" defaultMessage="When" />
+                    </div>
+                    <div className="dsl-editor__node">
+                        <ComponentSelection
```

Comment:
I noticed that if we have components with a long label, the dropdown becomes very long (I had one of these components because I was looking at bug https://github.com/open-formulieren/open-forms/issues/576 :smiley: )
![Screenshot from 2021-08-23 10-20-55](https://user-images.githubusercontent.com/19154114/130415022-a3b3b957-8018-4fe0-88ac-d16af5da4553.png)